casadm/cas_lib.c

```diff
@@ -1803,58 +1803,60 @@ int add_core(unsigned int cache_id, unsigned int core_id, const char *core_devic
 	return SUCCESS;
 }
 
-int _check_if_mounted(int cache_id, int core_id)
+bool _device_mounts_detected(int cache_id, int core_id)
 {
 	FILE *mtab;
 	struct mntent *mstruct;
 	char dev_buf[80];
-	int difference = 0, error = 0;
+	int no_match = 0, error = 0, cmplen = 0;
 	if (core_id >= 0) {
 		/* verify if specific core is mounted */
-		snprintf(dev_buf, sizeof(dev_buf), "/dev/cas%d-%d", cache_id, core_id);
+		cmplen = snprintf(dev_buf, sizeof(dev_buf), "/dev/cas%d-%d", cache_id, core_id);
 	} else {
-		/* verify if any core from given cache is mounted */
-		snprintf(dev_buf, sizeof(dev_buf), "/dev/cas%d-", cache_id);
+		/* verify if any core from given cache is mounted
+		   do not compare terminating NULL for cache */
+		cmplen = snprintf(dev_buf, sizeof(dev_buf), "/dev/cas%d-", cache_id) - 1;
 	}
 
 	mtab = setmntent("/etc/mtab", "r");
-	if (!mtab)
-	{
-		cas_printf(LOG_ERR, "Error while accessing /etc/mtab\n");
-		return FAILURE;
+	if (!mtab) {
+		/* if /etc/mtab not found then the kernel will check for mounts */
+		return false;
```

Comment:
If we can identify that the device is mounted, then we don't even try removing/stopping it, otherwise we try executing operation and return external error if it fails. For this particular case - line 1915 prints the error message and the external error (`Device mounts are pending`).